### PR TITLE
RUN-1924: Server identity UI updates

### PR DESCRIPTION
**Pull request by gschueler**
**Is this a bugfix, or an enhancement? Please describe.**

* replace direct dom loading of server-identity component with ui-socket usage
* fix some issues related to this change
    * navbar additional provides for vue
    * issues with ui-socket without normal context data (login page footer)
* minor cleanup
* conversion to typescript

### CODE REVIEW

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/server-identity/serverIdentity.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <server-display :uuid="serverInfo.uuid"
+                    :name="serverInfo.name"
+                    :name-class="nameClass"
+                    :glyphicon="serverInfo.icon"
+                    :show-id="showId"
+                    v-if="serverInfo">
+    </server-display>`
+})
+
+let rundeckContext = getRundeckContext()
```
Comment (JoseOrtiz) [inline rundeckapp/grails-spa/packages/ui-trellis/src/app/components/server-identity/serverIdentity.ts:37]: As this variable is not going to be reassigned is better to set it as `const`
```suggestion
const rundeckContext = getRundeckContext()
```

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/server-identity/serverIdentity.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      serverInfo: null
+    }
+  },
+  props: {
+    showId: {type: Boolean, default: true},
+    nameClass: {type: String, default: ''},
+    serverName: {type: String, default: ''},
+    serverUuid: {type: String, default: ''}
+  },
+  async mounted() {
```
Comment (JoseOrtiz) [inline rundeckapp/grails-spa/packages/ui-trellis/src/app/components/server-identity/serverIdentity.ts:24]: There is no `await` method to be awaited, so you could remove the `async`

Comment (gschueler) [inline rundeckapp/grails-spa/packages/ui-trellis/src/app/components/server-identity/serverIdentity.ts:24]:
```suggestion
 mounted() {
```

**Review by JoseOrtiz — APPROVED** [review]
Code looks good

**Pull request merged**